File: Java/Rectangulo/rectangulo/Rectangulo.java
package rectangulo;

public class Rectangulo {
  protected int alto;
  protected int ancho;

  // Constructor
  Rectangulo(int alto, int ancho) {
    this.alto = alto;
    this.ancho = ancho;
  }

  /**
   * Devuelve el valor del alto del rectángulo.
   * 
   * @return alto
   */
  int getAlto() {
    return alto;
  }

  /**
   * Asigna el valor del alto al rectángulo.
   * 
   * @param alto
   * @throws ArithmeticException
   */
  void setAlto(int alto) throws ArithmeticException {
    if (alto < 1)
      throw new ArithmeticException(
        "\nEl alto del rectángulo no puede ser menor que 1.");

    if (alto > 10)
      throw new ArithmeticException(
          "\nEl alto del rectángulo no puede ser mayor que 10.");

    this.alto = alto;
  }

  /**
   * Devuelve el valor del ancho del rectángulo.
   * 
   * @return
   */
  int getAncho() {
    return ancho;
  }

  /**
   * Asigna el valor del ancho al rectángulo.
   * 
   * @param ancho
   * @throws ArithmeticException
   */
  void setAncho(int ancho) throws ArithmeticException {
    if (ancho < 0)
      throw new ArithmeticException(
          "\nEl alto del rectángulo no puede ser menor que 0.");

    if (ancho > 10)
      throw new ArithmeticException(
          "\nEl alto del rectángulo no puede ser mayor que 10.");
    this.ancho = ancho;
  }

  /**
   * Muestra el rectángulo por pantalla.
   */
  public String toString() {
    String ancho = ("");
    String alto = ("");
    String resultado = ("");

    for (int i = 0; i < getAncho(); i++) {
      ancho += ("* ");

      if (i == 0 || i == getAncho() - 1) {
        alto += ("* ");
      } else {
        alto += ("  ");
      }
    }

    if (getAlto() > 1 || getAncho() > 1) {
      resultado += (ancho + "\n");
      for (int i = 0; i < getAlto() - 2; i++) {
        resultado += (alto + "\n");
      }
      if (getAlto() > 1) {
        resultado += ancho;
      }
    } else {
      resultado = ("*");
    }

    return resultado;
  }
}

File: Python/Rectangulo/Rectangulo.py
class Rectangulo():
  
  alto = 0
  ancho = 0
  
  def __init__(self, alto, ancho):
    self.alto = alto;
    self.ancho = ancho;
  
  # Devuelve el valor del alto del rectángulo.
  def getAlto(self):
    return self
  
  
  # Asigna el valor del alto al rectángulo.
  def setAlto(self, alto):
    try:
      if self.alto > 10 | self.alto < 0:
        raise TypeError("Valor no válido.")
        self.alto = alto
    
    except:
      print("Número no válido.")
  
  
  # Devuelve el valor del ancho del rectángulo.
  def getAncho(self):
    return self
  
  
  # Asigna el valor del ancho al rectángulo.
  def setAncho(self, ancho):
    try:
      if self.ancho > 10 | self.ancho < 1:
        raise TypeError("Valor no válido.")
        self.ancho = ancho
      
    except TypeError:
      print("Número no válido.")
      
  
  def __str__(self):
    altura = ""
    anchura = ""
    
    for num in range(self.ancho):
      anchura += ("* ")
      anchura += ("  ")
      
      if num == 0 | num == (self.ancho - 1):
        altura += ("* ")
        
      else:
        altura += ("  ")
      
    if self.alto > 1 | self.ancho > 1:
      print(anchura)
      for num in range(self.alto - 2):
        print(altura)
      
      if self.alto > 1:
        print(anchura)
    else:
      print("*")
      
